src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c

```diff
@@ -267,7 +268,7 @@ xocl_read_axlf_helper(struct xocl_drm *drm_p, struct drm_xocl_axlf *axlf_ptr)
 	}
 
 	xclbin_id = XOCL_XCLBIN_ID(xdev);
-	if (uuid_equal(xclbin_id, &bin_obj.m_header.uuid)) {
+	if (xclbin_id && uuid_equal(xclbin_id, &bin_obj.m_header.uuid)) {
```

Comment:
what if on a non-versal board and xclbin_id != bin_obj uuid?

Reply:
Did you see any different behavior on non-versal board by this change?

Reply:
ok, my description is in-correct. xclbin_id is actually a pointer. I am fine with it.